src/FxCop/Desktop.Analyzers/Core/Security/DoNotUseInsecureCryptographicAlgorithmsAnalyzer.cs

```diff
@@ -0,0 +1,185 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+using Desktop.Analyzers.Common;
+
+
+namespace Desktop.Analyzers
+{
+    public abstract class DoNotUseInsecureCryptographicAlgorithmsAnalyzer : DiagnosticAnalyzer
+    {
+        internal const string DoNotUseMD5RuleId = "CA5350";
```

Comment:
At one time, we'd discussed collapsing these checks into a single ID for a 'DoNotUseCryptographicallyWeakAlgorithms'. I think there's still merit to this position. @genlu, maybe we can discuss offline?